functions/types.go

```diff
@@ -0,0 +1,124 @@
+package functions
+
+import (
+	substraitgo "github.com/substrait-io/substrait-go"
+	"github.com/substrait-io/substrait-go/types"
+	"strings"
+)
+
```

Comment:
With more understanding of type system today, I feel "typeInfo" should represent type in the system (substrait/dialect) it is representing. That way it feels much better separation and extensible in future too



```
// This represent property of a type
type typeInfo struct {
	name         string
	supportedAsColumn bool
        nullabilityCharacter string
        parameterContainer string
}
```

TypeRegistry will have a name to type info map. This is information about property of a type in substrait

Reply:
Resolving since I had a thought that we need to store typeInfo for substrait too in LocalTypeRegistry. Because of [this](https://github.com/substrait-io/substrait-go/pull/32#issuecomment-2275364351) comment